src/full/Agda/Interaction/Imports.hs

```diff
@@ -717,9 +717,9 @@ getStoredInterface x file msrc = do
 
 -- | Report those given warnings that come from the given module.
 
-reportWarningsForModule :: MonadDebug m => TopLevelModuleName -> [TCWarning] -> m ()
+reportWarningsForModule :: MonadDebug m => TopLevelModuleName -> Set TCWarning -> m ()
 reportWarningsForModule x warns = do
-  unlessNull (filter ((Strict.Just (Just x) ==) . fmap rangeFileName . tcWarningOrigin) warns) \ ws ->
+  unlessNull (filter ((Strict.Just (Just x) ==) . fmap rangeFileName . tcWarningOrigin) $ Set.toList warns) \ ws ->
```

Comment:
```suggestion
  unlessNull (Set.filter ((Strict.Just (Just x) ==) . fmap rangeFileName . tcWarningOrigin) warns) \ ws ->
```

Reply:
Thanks for the review @gallais!

So here I frankly do not know what is better, `List.filter` or `Set.filter`.  The result `ws` is then only used for printing, iterated through, so the list structure is the most natural there.  
Since `Set.filter` needs to produce a balanced tree, there might be some wasted effort of rebalancing.  In the worst case, `Set.filter f` is `Set.fromList . List.filter f . Set.toList`.

Reply:
Ah yes, my bad: I did not notice `unlessNull` actually had a continuation to run.